Add modes and failsafe buttons

shadwstalkr opened the pull request:
This change adds all the modes and Shuffleboard buttons for failsafe modes.

shadwstalkr committed:
Add a mode for PID driving

This adds a mode for PID driving. This also adds the driver and gunner mode to the Shuffleboard display.

shadwstalkr committed:
Merge branch 'master' into add/driver-pid-mode

shadwstalkr committed:
Add modes

This adds all the gunner modes.

The mode manager supports a dedicated failsafe mode. The mode manager class hierarchy was expanded to simplify usage.

Failsafe buttons were added to the Shuffleboard test tab.

coderkevin reviewed (approved):
Reviewed in person, looks good.

coderkevin committed:
Simplify dpadmanager

This simplifies the dpad manager state, depending more on the base
manager class state.

coderkevin reviewed (commented):
Reviewed in-person with Robert
